### PR TITLE
Remove datadir arg

### PR DESCRIPTION
**Pull request by jeromekelleher**
This PR removes the mandatory DATADIR argument from the server CLI, and moves all configuration into the flask config space. This is an essential step towards deployability.

For development, we add a default configuration that we invoke from the CLI that makes the default data source ga4gh-example-data. This should make developers lives easy. We can override this with a config file specified on the command line though.

This will be updated and rebased once #212 has been merged. Feedback welcomed!

### CODE REVIEW

File: ga4gh/cli.py [hunk 1]
```diff
@@ -243,26 +241,11 @@ def addGlobalOptions(parser):
         "--port", "-P", default=8000, type=int,
         help="The port to listen on")
     parser.add_argument(
-        "--config", "-C", default='DefaultConfig', type=str,
+        "--config", "-c", default='DevelopmentConfig', type=str,
```
Comment (dcolligan) [inline ga4gh/cli.py]: why the changes here and below of the short args to lowercase?

**Comment by dcolligan** [comment]
Looks good, although I will want this to land after #212 since rebasing on top of that change will alter this one significantly.

File: ga4gh/cli.py (hunk 1 above)
Comment (jeromekelleher) [inline ga4gh/cli.py]: I think we usually fill up the lowercase letters first when making options. I'd imagine -f is a bit more common for filename input than -F. I'm happy to change it back if you don't like it, I'm not that bothered one way or the other.

**Comment by jeromekelleher** [comment]
Good, glad you like it. We definitely want to land #212 first; I thought this was the best way to explain where I imagined our input was going after we merged the simulators!

File: ga4gh/cli.py (hunk 1 above)
Comment (dcolligan) [inline ga4gh/cli.py]: No, this is fine.  Just wondered what the rationale was.

**Comment by jeromekelleher** [comment]
OK, this has been updated with the simulated backend and is ready for review.

One problem is the README is now out of date. I don't think there's much point in patching it up quickly here though, as we need to rework it substantially to start removing the wormtable vs tabix distinction, and to start documenting the config file format. This should be done separately I think - I'll open an issue if this PR is merged.

File: ga4gh/backend.py [hunk 2]
```diff
@@ -292,6 +292,12 @@ def setResponseValidation(self, responseValidation):
         self._responseValidation = responseValidation
 
 
+class EmptyBackend(AbstractBackend):
```
Comment (dcolligan) [inline ga4gh/backend.py]: What is the motivation behind EmptyBackend?

Comment (jeromekelleher) [inline ga4gh/backend.py]: I thought it would be useful for testing - there are times when we want to assert some properties of the backend when we don't care about data at all (`test_views` uses it now I think). I didn't want to instantiate `AbstractBackend` directly as this feels a bit wrong, and we might change the behaviour of `AbstractBackend` later to now be directly instantiable.

Comment (dcolligan) [inline ga4gh/backend.py]: Cool, I'm +1 on these.

**Comment by macieksmuga** [comment]
Ah, flask config file. It's rather hard to figure out how to run the server now, for a casual user. README desperately needs a rewrite.
